jbpm-wb-process-runtime/jbpm-wb-process-runtime-client/src/main/java/org/jbpm/workbench/pr/client/editors/instance/diagram/ProcessInstanceDiagramPresenter.java

```diff
@@ -209,6 +210,20 @@ public void onDiagramNodeSelected(final String nodeId) {
         }
     }
 
+    public void onDiagramNodeSelectedByDblClick(final String nodeId) {
+        Optional<NodeInstanceSummary> nodeInstance = nodeInstances.stream().filter(node -> node.getNodeUniqueName().equals(nodeId)).findFirst();
```

Comment:
@bxf12315 here, instead of getting just the first instance we need to check if there are more than one.

If the count > 1 then disallow navigation and show alert instead
If the count==1 navigate to the sub-process instance.

So below, instead of nodeInstance.isPresent() we can just check the count of elements. In the stream above, instead of findFirst(), use count() and get the number to compare later.